### PR TITLE
Fix #19832: Scenery Search

### LINKED ISSUE #19832: Scenery search
### Operating System

Windows 10 x64

### OpenRCT2 build

v0.4.4-81fd7b6

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

_No response_

### Describe the issue

Using the new search feature in the scenery tab.  After adding scenery objects via the Cheat menu, I try to add scenery to the map.  

The search field doesn't seem to find all relevant items based on the in-game description of the object.  

### Steps to reproduce

In the scenery tab: 
Click on a scenery group tab ("Signs And Items For Footpaths") and search for "sign":
SignsAndItemsForFootpaths

Then click on the "All Scenery" group tab and search for the same:
AllScenery

Search results are different.  Note the scrollbar in each screenshot - all found items are displayed.

I do have lots of objects in the Object folder that I've collected over the years.  I'm not sure if there are too many for the index to capture?  

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by trzejos**
An early return in `InitSceneryEntry` was causing some scenery items to be excluded from the all scenery tab. This reorders that method so that entries are always added to the all scenery tab first.

image

### CODE REVIEW

File: distribution/changelog.txt [hunk 1]
```diff
@@ -31,6 +31,7 @@
 - Fix: [#19767] No message when path is not connected to ride exit and is therefore unreachable for mechanics.
 - Fix: [#19800] Crash when displaying station stats with more than 62 stations.
 - Fix: [#19801] The in-game load/save window cannot be resized anymore.
+- Fix: [#19832] Scenery Search
```
Comment (tupaschoal) [inline distribution/changelog.txt:34]: Looks a bit too short

**Review by tupaschoal — APPROVED** [review]
Looks sensible, but haven't tested

File: distribution/changelog.txt (hunk 1 above)
Comment (trzejos) [reply distribution/changelog.txt:34]: I titled it after the open issue that this addresses. Would you prefer something more descriptive?

Comment (trzejos) [reply distribution/changelog.txt]: I've updated the changelog entry.

File: src/openrct2-ui/windows/Scenery.cpp [hunk 2]
```diff
@@ -1128,8 +1128,12 @@ class SceneryWindow final : public Window
 
         if (IsSceneryAvailableToBuild(selection))
         {
-            // Get current tab
-            const auto tabIndex = FindTabWithScenery(selection);
+            // Add scenery to all tab
+            auto allTabInfo = GetSceneryTabInfoForAll();
```
Comment (ZehMatt) [inline src/openrct2-ui/windows/Scenery.cpp:1132]:
```suggestion
            auto* allTabInfo = GetSceneryTabInfoForAll();
```

**Review by ZehMatt — CHANGES_REQUESTED** [review]
One minor thing, change seems sensible.

**Review by ZehMatt — APPROVED** [review]

**Comment by deurklink** [comment]
I saw some people complain that this PR may have caused their Misc tab (the question mark) to disappear

**Comment by trzejos** [comment]
@deurklink can you open a new issue on this with an example save and I'll look into it.

**Comment by ZehMatt** [comment]
> @deurklink can you open a new issue on this with an example save and I'll look into it.

Already took care of it.

**Pull request merged**